src/uuid.ts

```diff
@@ -1,12 +1,39 @@
+import { intRangeWithEngine } from "./int-range.js";
 import { Engine, defaultEngine } from "./utils.js"
 
 const uuid = (engine: Engine = defaultEngine): string => {
   // Credit @Alexey Silin from https://gist.github.com/1308368
-  let a: any = ''
-  let b: any = ''
-  // eslint-disable-next-line
-  for (b = a = ''; a++ < 36; b += ~a % 5 | a * 3 & 4 ? (a ^ 15 ? 8 ^ engine() * (a ^ 20 ? 16 : 4) : 4).toString(16) : '-') { }
-  return b
+	// Creates a random v4 UUID of the form xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx,
+	// where each x is a random hexadecimal digit from 0 to f, and y is a random
+	// hexadecimal digit from 8 to b.
+	const randomRange = intRangeWithEngine(engine)
+	const defaultGenerator = (): string => randomRange(0, 15).toString(16)
+	const groups: object[] = [
```

Comment:
prob want this to be more specific to avoid linter issues:
```es6
type Group = {
  length: number;
  generator?: (index: number) => number | string
}

const groups: Group[]
```